### PR TITLE
Add support for raw string with multi '#'s(Issue 815)

### PR DESCRIPTION
**Pull request by kngwyu**
Solves #815.
Add support for nested raw string like 
```rust
 r###" "" r#""# "###
```
to```CodeIndiceIter```.

### CODE REVIEW

File: src/racer/codecleaner.rs [hunk 1]
```diff
@@ -1,11 +1,20 @@
 use core::{Point, SourceByteRange};
 
+/// Type of the string
+#[derive(Clone, Copy, Debug)]
+enum StringType {
+    /// Raw(n) => raw string started with n #s
+    Raw(usize),
+    /// normal string starts with "
+    NotRaw,
```
Comment (TedDriggs) [inline src/racer/codecleaner.rs]: Please use `Cooked` to match [`syn`'s convention](https://docs.rs/syn/0.12.10/syn/enum.StrStyle.html)

File: src/racer/codecleaner.rs [hunk 2]
```diff
@@ -123,26 +133,58 @@ impl<'a> CodeIndicesIter<'a> {
...
         let mut pos = self.pos;
-        if pos > 1 && src_bytes[pos-2] == b'r' {
-            // raw string (eg br"\"): no escape
-            match src_bytes[pos..].iter().position(|&b| b == b'"') {
-                Some(p) => pos += p+1,
-                None    => pos = src_bytes.len()
+        match str_type {
+            StringType::Raw(level) => {
+                // raw string (eg br#"\"#)
+                // detect corresponding end(if start is r##", ##") greedily
```
Comment (TedDriggs) [inline src/racer/codecleaner.rs]: `if start is r##", ##"` - should the second one of those be `"##`, or am I misunderstanding this comment?

File: src/racer/codecleaner.rs [hunk 3]
```diff
@@ -123,26 +133,58 @@ impl<'a> CodeIndicesIter<'a> {
...
+                enum SharpState {
+                    Sharp((usize, usize)), // (Num of preceeding #s, Pos of end ")
```
Comment (TedDriggs) [inline src/racer/codecleaner.rs]: Nitpick: `preceeding` should be `preceding` (same goes for line 145)

**Comment by kngwyu** [comment]
Thanks for review. Fixed.

**Comment by nrc** [comment]
Thanks for addressing the review comments!

**Pull request merged**